docker/test/base/Dockerfile

```diff
@@ -52,14 +52,18 @@ RUN apt-get update \
         --yes --no-install-recommends
 
 # Sanitizer options for services (clickhouse-server)
-RUN echo "TSAN_OPTIONS='verbosity=1000 halt_on_error=1 history_size=7'" >> /etc/environment; \
+# Set resident memory limit for TSAN to 50GiB (51200MiB) to avoid OOMs in Stress tests
```

Comment:
I've tried it in #30009, but it does not helps.
What does helps is #30579, it does helps, but not completelly.
I've looked into OOM in TSan after #30579 but it does not looks like recursion problem anymore, and recursion limit is already small enough, so I've postpone it for awhile.